### PR TITLE
579 create suggestion instead duplicate

### PR DESCRIPTION
**Pull request by ahakanzn**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -31,7 +31,7 @@
     <okio.version>2.4.3</okio.version>
     <lombok.version>1.18.12</lombok.version>
     <jcommander.version>1.78</jcommander.version>
-    <gbif-api.version>1.16.3</gbif-api.version>
+    <gbif-api.version>1.17.0-SNAPSHOT</gbif-api.version>
```
Comment (marcos-lg) [inline pom.xml:34]: If this is a jdk11 version this project should be updated to use java11.

File: src/main/java/org/gbif/collections/sync/SyncResultExporter.java [hunk 2]
```diff
@@ -135,6 +133,16 @@ public static void exportResultsToFile(SyncResult result, Path filePath) {
     }
   }
 
+  private static void printSuggestedEntity(BufferedWriter writer, int key){
+    try {
```
Comment (marcos-lg) [inline src/main/java/org/gbif/collections/sync/SyncResultExporter.java:137]: Shouldn't this print the whole entity and not just the key?

File: src/main/java/org/gbif/collections/sync/common/BaseSynchronizer.java [hunk 3]
```diff
@@ -1,24 +1,29 @@
...
-import java.util.List;
-
-import com.google.common.annotations.VisibleForTesting;
-
 public abstract class BaseSynchronizer<S, R> {
 
   protected final GrSciCollProxyClient proxyClient;
   protected final StaffResultHandler<S, R> staffResultHandler;
   protected final EntityConverter<S, R> entityConverter;
+  private final String COMMENT = "This suggestion was created as part of the weekly synchronisation"
```
Comment (marcos-lg) [inline src/main/java/org/gbif/collections/sync/common/BaseSynchronizer.java:25]: shouldn't it also be static?

File: src/main/java/org/gbif/collections/sync/common/BaseSynchronizer.java [hunk 4]
```diff
@@ -126,32 +131,61 @@ public InstitutionAndCollectionMatch handleInstAndCollMatch(MatchResult<S, R> ma
...
+      return NoEntityMatch.builder().build();
+    }
+    //Check if there is another institution with the same name
+    List<Institution> institutionWithSameName = proxyClient.findInstitutionByName(newInstitution.getName());
+    //If not, we should create a suggestion with the option of creating an institution suggestion
+    if (institutionWithSameName.isEmpty()) {
+      return createAndSuggestCollection(matchResult, newInstitution, ihIdentifiers, true);
+    }
+    else {
+      return createAndSuggestCollection(matchResult, institutionWithSameName.get(0), ihIdentifiers, false);
```
Comment (marcos-lg) [inline src/main/java/org/gbif/collections/sync/common/BaseSynchronizer.java:148]: it should still use the `newInstitution`. If you use `institutionWithSameName.get(0)`you are duplicating it.

File: src/main/java/org/gbif/collections/sync/SyncResultExporter.java (hunk 2 above)
Comment (ahakanzn) [reply src/main/java/org/gbif/collections/sync/SyncResultExporter.java:137]: creating a suggested entity endpoint in the registry returns only the key, should I modify it to return the whole entity? So I can use it here? Or do you think I should go to registry again with the keys and get the entities?

Comment (marcos-lg) [reply src/main/java/org/gbif/collections/sync/SyncResultExporter.java:137]: I just saw the rest of the PR, you should store the whole suggestion in the NoEntityMatch

File: src/main/java/org/gbif/collections/sync/common/BaseSynchronizer.java (hunk 4 above)
Comment (ahakanzn) [reply src/main/java/org/gbif/collections/sync/common/BaseSynchronizer.java:148]: I think so as well, I'll ask her again when she comes back

File: src/main/java/org/gbif/collections/sync/SyncResultExporter.java (hunk 2 above)
Comment (ahakanzn) [reply src/main/java/org/gbif/collections/sync/SyncResultExporter.java:137]: Okay makes sense

**Pull request merged**